do-catalog/src/store/actions/catalog.js

```diff
@@ -0,0 +1,18 @@
+export function fetchDataset (context, datasetId) {
+  context.commit('setFetchingState');
+
+  var baseUrl = "https://public.carto.com";
+  var apiPath = "/api/v4/data/observatory/metadata/datasets/";
+  var url = baseUrl + apiPath + datasetId;
+  var req = new XMLHttpRequest();
+  req.open('GET', url, true);
+  req.onreadystatechange = function () {
```

Comment:
The good old classic way. No need to change it yet, as we will try to use the same method as in dashboard, but you can use `fetch` to ease the calls for following cases.